.storybook/main.js

```diff
@@ -3,40 +3,31 @@ const fs = require('fs');
 const { TsconfigPathsPlugin } = require('tsconfig-paths-webpack-plugin');
 
 /**
- *  @callback StorybookWebpackConfig
- *  @param {import("webpack").Configuration} config
- *  @param {{configType: 'DEVELOPMENT' | 'PRODUCTION'}} options - change the build configuration. 'PRODUCTION' is used when building the static version of storybook.
- *  @returns {import("webpack").Configuration}
- */
-
-/**
- *  @typedef {{
- *    check:boolean;
- *    checkOptions: Record<string,unknown>;
- *    reactDocgen: string | boolean;
- *    reactDocgenTypescriptOptions: Record<string,unknown>
- *  }} StorybookTsConfig
- */
-
-/**
- *  @typedef {{
- *    stories: string[];
- *    addons: string[];
- *    typescript: StorybookTsConfig;
- *    babel: (options:Record<string,unknown>)=>Promise<Record<string,unknown>>;
- *    webpackFinal: StorybookWebpackConfig;
- *    core: {builder:'webpack5'};
- *    previewHead: (head: string) => string
- * }} StorybookConfig
+ * @typedef {import('@storybook/core-common').StorybookConfig} StorybookBaseConfig
```

Comment:
I updated this to use types from storybook itself (they might have gotten more detailed than they were in the past)